### PR TITLE
Adding controller and spec for template configuration

### PR DESCRIPTION
**Pull request by RodrigoMedinar**

### CODE REVIEW

File: Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /*cd Ma
+    @PostMapping("/api/foos")
+@ResponseBody
+public String addFoo(@RequestParam(name = "id") String fooId, @RequestParam String name) {
+    return "ID: " + fooId + " Name: " + name;
+}
+     */
+    @PreAuthorize("isAdmin()")
```
Comment (terrypacker) [inline Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java:58]: Annotate with `@ApiOperation` for Swagger.  Take a look at other controllers.

File: Mango API/src/com/infiniteautomation/mango/rest/latest/ServerRestController.java [hunk 2]
```diff
@@ -291,6 +291,7 @@ public ResponseEntity<List<SessionInformation>> listSessions(@AuthenticationPrin
         @ApiResponse(code = 500, message = "Internal error", response = ResponseEntity.class),})
     @RequestMapping(method = {RequestMethod.GET}, value = "system-info")
     public ResponseEntity<Map<String, Object>> getSystemInfo(@AuthenticationPrincipal PermissionHolder user) {
+        System.out.println("Testing on the fly thing");
```
Comment (terrypacker) [inline Mango API/src/com/infiniteautomation/mango/rest/latest/ServerRestController.java:294]: Revert all changes to this class.  When done we will squash this into a single commit to remove this change entirely.

**Review by terrypacker — CHANGES_REQUESTED** [review]
This PR should be the second one to be merged as it will depend on the Service.  So make a PR to get the service into 5.0.x because if we merge this it won't build.

Also, to have Team City builds pass you need the same branch on all the repositories so you can use git submodule commands (found in the IDE setup document) to create branches in all the repos.  Then you only make the PRs for the branch(es) you want to merged.  Once the PRs are merged you can delete all the other branches manually using git submodule commands.

File: Mango API/src/com/infiniteautomation/mango/rest/latest/ServerRestController.java (hunk 2 above)
Comment (RodrigoMedinar) [reply Mango API/src/com/infiniteautomation/mango/rest/latest/ServerRestController.java:294]: changes removed.

File: Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java (hunk 1 above)
Comment (RodrigoMedinar) [reply Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java:58]: fixed.

File: Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import com.serotonin.m2m2.Common;
+import com.serotonin.m2m2.i18n.TranslatableMessage;
+import com.serotonin.m2m2.vo.permission.PermissionHolder;
+
+import io.swagger.annotations.Api;
+
+/**
+ * Class to provide server information
+ *
+ * @author Terry Packer
```
[the remark this answers is not in the document]
Comment (RodrigoMedinar) [reply Mango API/src/com/infiniteautomation/mango/rest/latest/ConfigurationTemplateRestController.java:36]: fixed

**Pull request closed without merging**